File: LeetCode/2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        unordered_map<int, int> mp;
        vector<int> ans;
        int n = A.size();
        int count = 0;
        for (int i = 0; i < n; i++) {
            mp[A[i]]++;
            mp[B[i]]++;
            if (A[i] == B[i]) {
                count++;
            } else {
                if (mp[A[i]] == 2) count++;
                if (mp[B[i]] == 2) count++;
            }
            ans.push_back(count);
        }
        return ans;
    }
};

File: LeetCode/2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        Map<Integer, Integer> map = new HashMap<>();
        int n = A.length;
        int[] result = new int[n];
        int count = 0;
        for (int i = 0; i < n; i++) {
            map.put(A[i], map.getOrDefault(A[i], 0) + 1);
            map.put(B[i], map.getOrDefault(B[i], 0) + 1);
            if (A[i] == B[i]) {
                count++;
            } else {
                if (map.get(A[i]) == 2) count++;
                if (map.get(B[i]) == 2) count++;
            }
            result[i] = count;
        }
        return result;
    }
}